engine-nextgen/src/test/kotlin/graphql/nadel/validation/NadelInputValidationTest.kt

```diff
@@ -0,0 +1,226 @@
+                            P3
+                            P4
+                            P5
+                            P6
+                        }
+                    """.trimIndent(),
+                ),
+            )
+
+            val errors = validate(fixture)
+            assert(errors.map { it.message }.isEmpty())
+        }
+
+        it("passes if input type was renamed") {
+            val fixture = NadelValidationTestFixture(
+                overallSchema = mapOf(
+                    "shared" to renameDirectiveDef,
+                    "test" to """
+                        type Query {
+                            pay(role: Role): Int
+                        }
+                        input Role @renamed(from: "PayFilter") {
+                            p: Professional
+                            m: Manager
+                        }
+                        enum Manager {
+                            M6 M4 M3 M5
+                        }
+                        enum Professional {
+                            P5 P4 P6 P3
+                        }
+                    """.trimIndent(),
+                ),
+                underlyingSchema = mapOf(
+                    "shared" to """
+                        scalar Nothing
+                        type Query {
+                            nothing: Nothing
+                        }
+                    """.trimIndent(),
+                    "test" to """
+                        type Query {
+                            pay(role: PayFilter): Int
+                        }
+                        input PayFilter {
+                            p: Professional
+                            m: Manager
+                        }
+                        enum Manager {
+                            M6 M4 M3 M5
+                        }
+                        enum Professional {
+                            P5 P4 P6 P3
+                        }
+                    """.trimIndent(),
+                ),
+            )
+
+            val errors = validate(fixture)
+            assert(errors.map { it.message }.isEmpty())
+        }
+
+        it("fails if input field in renamed type does not exist in underlying type") {
+            val fixture = NadelValidationTestFixture(
+                overallSchema = mapOf(
+                    "shared" to renameDirectiveDef,
+                    "test" to """
+                        type Query {
+                            pay(role: Role): Int
+                        }
+                        input Role @renamed(from: "PayFilter") {
+                            p: Professional
+                            m: Manager
+                        }
+                        enum Manager {
+                            M6 M4 M3 M5
+                        }
+                        enum Professional {
+                            P5 P4 P6 P3
+                        }
+                    """.trimIndent(),
```

Comment:
Yeah great clean test - real life data makes testing easier - rather than `Foo` and `Bar`